pkg/minishift/openshift/registry.go

```diff
@@ -25,6 +25,44 @@ import (
 )
 
 func GetDockerRegistryInfo() (string, error) {
+	var registryInfo string
+	var err error
+	if isRegistryRouteEnable() {
+		registryInfo, err = fetchRegistryRoute()
+	} else {
+		registryInfo, err = fetchRegistryService()
+	}
+	return registryInfo, err
+}
+
+func isRegistryRouteEnable() bool {
```

Comment:
@hferentschik I made that change but as I suspected if a user use `addon apply` than it is not going to work :(,